### PR TITLE
Add camel-servicenow wrapper

### PR DESCRIPTION
**Pull request by lopushen**
used `camel-cxf` feature to cover most of the dependencies, including `camel-core`

### CODE REVIEW

**Comment by f2par0** [comment]
camel-cxf includes too many unnecessary dependencies : it works with 

   ```
 <feature name='camel-servicenow' version='${project.version}' start-level='50'>
        <feature version='${project.version}'>camel-core</feature>
        <feature version='${project.version}'>camel-blueprint</feature>
        <feature>jackson</feature>
        <feature>jakarta-jws</feature>
        <feature>jakarta-servlet</feature>
        <feature version="[11,12)">jetty</feature>
        <bundle>mvn:org.apache.camel.karaf/camel-cxf-all/${project.version}</bundle>
        <bundle>mvn:org.apache.camel.karaf/camel-attachments/${project.version}</bundle>
        <bundle>mvn:org.apache.camel.karaf/camel-http-base/${project.version}</bundle>
        <bundle dependency="true">mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.wsdl4j/1.6.3_1</bundle>
        <bundle dependency="true">mvn:org.apache.ws.xmlschema/xmlschema-core/2.3.1</bundle>
        <bundle dependency="true">mvn:org.apache.neethi/neethi/3.2.0</bundle>
        <bundle dependency="true">mvn:org.glassfish.hk2/osgi-resource-locator/2.5.0-b42</bundle>
        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jdk8/${jackson2-version}</bundle>
        <bundle dependency='true'>mvn:com.fasterxml.jackson.jakarta.rs/jackson-jakarta-rs-base/${jackson2-version}</bundle>
        <bundle dependency='true'>mvn:com.fasterxml.jackson.jakarta.rs/jackson-jakarta-rs-json-provider/${jackson2-version}</bundle>
        <bundle dependency='true'>mvn:org.apache.cxf/cxf-rt-rs-security-oauth2/${cxf-version}</bundle>
        <bundle dependency='true'>mvn:org.apache.cxf/cxf-rt-rs-json-basic/${cxf-version}</bundle>
        <bundle dependency='true'>mvn:org.apache.cxf/cxf-rt-rs-security-jose-jaxrs/${cxf-version}</bundle>
        <bundle dependency='true'>mvn:org.apache.cxf/cxf-rt-rs-security-jose/${cxf-version}</bundle>
        <bundle>mvn:org.apache.camel.karaf/camel-servicenow/${project.version}</bundle>
    </feature>
```
this means, camel-spring, jakarta-servlet, validation, camel-cxf-blueprint , servicemix cgi lib are not used

**Comment by essobedo** [comment]
Please, don't use the feature `camel-cxf` neither the `camel-cxf-all` wrapper, use directly the OSGI bundles of CXF, indeed `camel-cxf` brings a lot of useless dependencies in this case

**Comment by lopushen** [comment]
> <feature name='camel-servicenow' version='${project.version}' start-level='50'>
>      <feature version='${project.version}'>camel-core</feature>
>      <feature version='${project.version}'>camel-blueprint</feature>
>      <feature>jackson</feature>
>      <feature>jakarta-jws</feature>
>      <feature>jakarta-servlet</feature>
>      <feature version="[11,12)">jetty</feature>
>      <bundle>mvn:org.apache.camel.karaf/camel-cxf-all/${project.version}</bundle>
>      <bundle>mvn:org.apache.camel.karaf/camel-attachments/${project.version}</bundle>
>      <bundle>mvn:org.apache.camel.karaf/camel-http-base/${project.version}</bundle>
>      <bundle dependency="true">mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.wsdl4j/1.6.3_1</bundle>
>      <bundle dependency="true">mvn:org.apache.ws.xmlschema/xmlschema-core/2.3.1</bundle>
>      <bundle dependency="true">mvn:org.apache.neethi/neethi/3.2.0</bundle>
>      <bundle dependency="true">mvn:org.glassfish.hk2/osgi-resource-locator/2.5.0-b42</bundle>
>      <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
>      <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jdk8/${jackson2-version}</bundle>
>      <bundle dependency='true'>mvn:com.fasterxml.jackson.jakarta.rs/jackson-jakarta-rs-base/${jackson2-version}</bundle>
>      <bundle dependency='true'>mvn:com.fasterxml.jackson.jakarta.rs/jackson-jakarta-rs-json-provider/${jackson2-version}</bundle>
>      <bundle dependency='true'>mvn:org.apache.cxf/cxf-rt-rs-security-oauth2/${cxf-version}</bundle>
>      <bundle dependency='true'>mvn:org.apache.cxf/cxf-rt-rs-json-basic/${cxf-version}</bundle>
>      <bundle dependency='true'>mvn:org.apache.cxf/cxf-rt-rs-security-jose-jaxrs/${cxf-version}</bundle>
>      <bundle dependency='true'>mvn:org.apache.cxf/cxf-rt-rs-security-jose/${cxf-version}</bundle>
>      <bundle>mvn:org.apache.camel.karaf/camel-servicenow/${project.version}</bundle>
>  </feature>

That's exactly what I got from gradually adding dependencies, then I thought that all of them are already contained in camel-cxf, but indeed there are extra ones

Also, camel-blueprint brings additional bundles
we might use `<feature>aries-blueprint</feature>` instead, is that ok?

**Comment by essobedo** [comment]
As mentioned before please don't use the `camel-cxf-all` wrapper, use only CXF bundles

File: features/src/main/feature/camel-features.xml [hunk 1]
```diff
@@ -1733,20 +1733,26 @@
...
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jdk8/${jackson2-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.jaxrs/jackson-jaxrs-base/${jackson2-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/${jackson2-version}</bundle>-->
-<!--        <bundle>mvn:org.apache.camel/camel-servicenow/${project.version}</bundle>-->
-<!--    </feature>-->
+    <feature name='camel-servicenow' version='${project.version}' start-level='50'>
+        <feature version='${project.version}'>camel-core</feature>
+        <feature version='[2.16,2.17)'>jackson</feature>
+        <feature version="[5,6)">jakarta-servlet</feature>
```
Comment (lopushen) [inline features/src/main/feature/camel-features.xml:1739]: jakarta-servlet is not in the dependency tree, but is imported by cxf-rt-frontend-jaxrs

File: features/src/main/feature/camel-features.xml [hunk 2]
```diff
@@ -1733,20 +1733,26 @@
...
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-databind/${jackson2-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-annotations/${jackson2-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jdk8/${jackson2-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.jaxrs/jackson-jaxrs-base/${jackson2-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/${jackson2-version}</bundle>-->
-<!--        <bundle>mvn:org.apache.camel/camel-servicenow/${project.version}</bundle>-->
-<!--    </feature>-->
+    <feature name='camel-servicenow' version='${project.version}' start-level='50'>
+        <feature version='${project.version}'>camel-core</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='${camel.osgi.version.range}'>camel-core</feature>
```
Please use version range for features

**Review by essobedo — APPROVED** [review]
LGTM

**Pull request merged**